src/backend/stubs/controllers/users_controller.stub

```diff
@@ -3,30 +3,87 @@
 }}}
 import type { HttpContext } from '@adonisjs/core/http';
 
+import { DateTime } from 'luxon';
 import vine from '@vinejs/vine';
 import { UsersProps } from '@story-cms/kit';
 import User from '#models/user';
+import Activity from '#models/activity';
+import ActivityService from '#services/activity_service';
 import {
   createUserValidator,
   updateUserValidator,
   validationMessages,
 } from '#validators/user';
+import Page from '#models/page';
+import Ui from '#models/ui';
+import Draft from '#models/draft';
+import Chapter from '#models/chapter';
 
 export default class UsersController {
+  private async logActivity(
+    userId: number | undefined,
+    locale: string,
+    action: Activity['action'],
+  ) {
+    if (!userId) return;
+
+    const activity = new ActivityService();
+    await activity.logAction(userId, locale, 'user', action);
+  }
+
+  private async getLastActivity(userId: number): Promise<string> {
```

Comment:
Since we now have a user activity table, it should not be necessary to have an n+1 query here. How about one query to the database group by userId and aggregate max(created_at). Then we have the whole list in memory